src/aspire/source/__init__.py

```diff
@@ -526,24 +529,174 @@ def vol_forward(self, vol, start, num):
         im *= self.amplitudes[all_idx, np.newaxis, np.newaxis]
         return im
 
-    def save(self, starfile_filepath, batch_size=512, save_mode=None, overwrite=False):
+    def save(
+        self,
+        starfile_filepath,
+        batch_size=512,
+        save_images=True,
+        save_mode=None,
+        overwrite=False,
+    ):
         """
-        Save the output images to mrc files
+        Save the output metedata to STAR file and/or images to MRCS file
 
+        :param starfile_filepath: Path to STAR file where we want to
+            save metadata of image_source
         :param batch_size: Batch size of images to query.
+        :param save_images: Whether to save all images or not.
         :param save_mode: Whether to save all images in a single or multiple files in batch size.
         :param overwrite: Option to overwrite the output mrcs files.
         """
-        logger.info("save images")
+        logger.info("save metedata into STAR file")
+        self._save_metadata(
+            starfile_filepath, batch_size=batch_size, save_mode=save_mode
+        )
+        if save_images:
+            logger.info("save images into MRCS file")
+            self._save_images(
+                starfile_filepath,
+                batch_size=batch_size,
+                save_mode=save_mode,
+                overwrite=overwrite,
+            )
+
+    def _save_metadata(self, starfile_filepath, batch_size=512, save_mode=None):
```

Comment:
Need to have a flag here to of whether to generate new image filenames or use existing ones (which we would like to do if we have estimated orientations, for example).

Reply:
Done in commit c1dae2b.

Reply:
I don't see it. Is this what `save_images` is supposed to be?

Reply:
Yes. I use the `save_images`  as an option.

Reply:
I don't think that's a very descriptive name here. What about `new_mrcs` or `new_images`?

Reply:
Changed to `new_mrcs` in commit  2e2eaa9.